### PR TITLE
added auth0 integration for superadmins

### PR DESCRIPTION
**Pull request by benadida**
Same approach we've taken for OAuth URLs, now for superadmins, with its own independent tenant. Difference is this manages a separate session variable and sends the user to a different URL, `/superadmin/`.

### CODE REVIEW

File: arlo_server/routes.py [hunk 1]
```diff
@@ -1246,6 +1267,30 @@ def logout():
...
+    return auth0_sa.authorize_redirect(redirect_uri=redirect_uri)
+
+
+@app.route(SUPERADMIN_OAUTH_CALLBACK_URL)
+def superadmin_login_callback():
+    auth0_sa.authorize_access_token()
+    resp = auth0_sa.get("userinfo")
+    userinfo = resp.json()
+
+    # we rely on the auth0 auth here, but check against a single approved domain.
```
Comment (jonahkagan) [inline arlo_server/routes.py:1282]: why is this necessary?

File: tests/routes_tests/test_auth.py [hunk 2]
```diff
@@ -93,6 +95,30 @@ def check_redirect_contains_redirect_uri(response, expected_url):
...
+
+    mock_response = Mock()
+    mock_response.json = MagicMock(return_value={"email": SA_EMAIL})
+    auth0_sa.get = Mock(return_value=mock_response)
+
+    rv = client.get("/auth/superadmin/callback?code=foobar")
+    assert rv.status_code == 302
+
+    with client.session_transaction() as session:  # type: ignore
+        assert session["_superadmin"]
```
Comment (jonahkagan) [inline tests/routes_tests/test_auth.py:116]: maybe some assertion about not messing with the rest of the session?

File: tests/routes_tests/test_auth.py [hunk 3]
```diff
@@ -93,6 +95,30 @@ def check_redirect_contains_redirect_uri(response, expected_url):
...
+
+    assert auth0_sa.authorize_access_token.called
+    assert auth0_sa.get.called
+
+
```
Comment (jonahkagan) [inline tests/routes_tests/test_auth.py:121]: update tests for logout?

File: arlo_server/auth.py [hunk 4]
```diff
@@ -53,7 +53,7 @@ def set_superadmin():
 
 
 def clear_superadmin():
-    del session["_superadmin"]  # pragma: no cover
+    session["_superadmin"] = False  # pragma: no cover
```
Comment (jonahkagan) [inline arlo_server/auth.py:56]: why this change?

File: arlo_server/routes.py (hunk 1 above)
Comment (benadida) [reply arlo_server/routes.py:1282]: defense-in-depth, I don't want to mistakenly get to a place where we forget to limit users in Auth0, and then anyone with a gmail address gets superadmin.
